server/api/auth/linkedin.js

```diff
@@ -0,0 +1,110 @@
+const passport = require('passport');
+var LinkedInStrategy = require('passport-linkedin-oauth2').Strategy;
+const loginWithIdp = require('./loginWithIdp');
+const { createIdToken } = require('../../api-util/idToken');
+
+const radix = 10;
+const PORT = parseInt(process.env.REACT_APP_DEV_API_SERVER_PORT, radix);
+const rootUrl = process.env.REACT_APP_CANONICAL_ROOT_URL;
+const clientID = process.env.REACT_APP_LINKEDIN_CLIENT_ID;
+const clientSecret = process.env.LINKEDIN_CLIENT_SECRET;
+
+let callbackURL = null;
+
+const useDevApiServer = process.env.NODE_ENV === 'development' && !!PORT;
+
+if (useDevApiServer) {
+  callbackURL = `http://localhost:${PORT}/api/auth/linkedin/callback`;
+} else {
+  callbackURL = `${rootUrl}/api/auth/linkedin/callback`;
+}
+
+const strategyOptions = {
+  clientID,
+  clientSecret,
+  callbackURL,
+  scope: ['r_emailaddress', 'r_liteprofile'],
+  passReqToCallback: true,
+};
+
+const verifyCallback = (req, accessToken, refreshToken, rawReturn, profile, done) => {
+  // We can can use util function to generate id token to match OIDC so that we can use
+  // our custom id provider in Flex
+
+  console.log('Profile', profile);
+
+  const locale = Object.keys(profile._json.firstName.localized)[0];
+
+  const firstName = profile._json.firstName.localized[locale];
+  const lastName = profile._json.lastName.localized[locale];
+  const email = profile.emails[0].value;
+
+  // LikedIn API doesn't return information if the email is verified or not directly.
+  // However, it seems that with OAUTH2 flow authentication is not possible if the email is not verified.
+  // There is no official documentation about this, but through testing it seems like this can be trusted
+  // For reference: https://stackoverflow.com/questions/19278201/oauth-request-verified-email-address-from-linkedin
+
+  const user = {
+    userId: profile.id,
+    profile: {
+      firstName,
+      lastName,
+      email,
+      emailVerified: true,
+    },
+  };
+
+  const state = req.query.state;
+  const queryParams = JSON.parse(state);
+
+  const { from, defaultReturn, defaultConfirm } = queryParams;
+
+  const idpToken = createIdToken(user)
+    .then(idpToken => {
+      const userData = {
+        email,
+        firstName,
+        lastName,
+        idpToken,
+        from,
+        defaultReturn,
+        defaultConfirm,
+      };
+
+      console.log('userData:', userData);
```

Comment:
Does this contain something that should not be saved to server logs in production?